include/CLI/TypeTools.hpp

```diff
@@ -150,5 +170,27 @@ bool lexical_cast(std::string input, T &output) {
     return !is.fail() && !is.rdbuf()->in_avail();
 }
 
+/// sum a vector of flag representations
+template <typename T,
+          enable_if_t<std::is_integral<T>::value && std::is_unsigned<T>::value, detail::enabler> = detail::dummy>
+void sum_flag_vector(const std::vector<std::string> &flags, T &output) {
+    int64_t count = std::accumulate(flags.begin(), flags.end(), int64_t{0}, [](int64_t cur, const std::string &flag) {
+        cur += (flag == "1") ? 1 : ((flag == "-1") ? (-1) : std::stoll(flag));
+        return cur;
+    });
+    output = (count > 0) ? static_cast<T>(count) : T{0};
+}
+
+/// sum a vector of flag representations
+template <typename T,
+          enable_if_t<std::is_integral<T>::value && std::is_signed<T>::value, detail::enabler> = detail::dummy>
+void sum_flag_vector(const std::vector<std::string> &flags, T &output) {
+    int64_t count = std::accumulate(flags.begin(), flags.end(), int64_t{0}, [](int64_t cur, const std::string &flag) {
+        cur += (flag == "1") ? 1 : ((flag == "-1") ? (-1) : std::stoll(flag));
```

Comment:
Why not this:

```suggestion
        cur += std::stoll(flag);
```

Is that for speed, perhaps?

Reply:
mostly,  there will be two very common strings with "1" being most common,  didn't see a reason to not take advantage of that fact.

Reply:
I would add a little comment saying that soma reader (like me) wound have to work though the logic just to find it is doing stol.